include/boost/geometry/algorithms/detail/check_iterator_range.hpp

```diff
@@ -11,7 +11,7 @@
 #define BOOST_GEOMETRY_ALGORITHMS_DETAIL_CHECK_ITERATOR_RANGE_HPP
 
 #include <boost/core/ignore_unused.hpp>
-
+#include <boost/geometry/strategies/intersection_result.hpp>
```

Comment:
This file doesn't seem to be the correct place for this utility.

Reply:
Maybe, but this include in need by `check_iterator_range_with_info` to define the return type 
`geometry::segment_intersection_points<point_type>`

Reply:
Yes, I know why it's here. My point was that I'm not sure whether or not this is the correct place.

Another way is to implement it differently, e.g. either of these solution would allow to not use the specific result type:
- store the state in a predicate and return this predicate, just like STL algorithms returns function objects in some cases
- take result type as output parameter
- define the return type as the first template parameter
- define the return type in predicate

Reply:
Now, I see your point, sure, I defined the return type as template parameter and removed the include.